### ISSUE TITLE
Unnecessary parentheses from function call with lambda

### ISSUE THREAD

**Issue by KotlinIsland** [pinterest/ktlint#1068]
## Expected Rule behavior
```kt
fun foo(fn: () -> Unit) = fn()

foo() { println("asdf") } // should report unnecessary parentheses
```
